GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java

```diff
@@ -0,0 +1,260 @@
+
+        //check and request for all necessary permissions
+        if (!Utils.hasUserGrantedNecessaryPermissions(this)) {
+            Utils.requestNecessaryPermissions(this);
+        }
+
+        if (!Utils.isGpsEnabled(locationManager)) {
+            setGpsEnabled();
+        }
+
+        //Choose a location api, hide the radio group and show startAndStopButton
+        apiRadioGroup.setOnCheckedChangeListener((RadioGroup group, int checkedId) -> {
+            locationApiVersion = checkedId == R.id.m_radio_button_FLP ?
+                    GpsDataCaptureService.LocationApiVersion.FUSEDLOCATIONPROVIDERCLIENT
+                    : GpsDataCaptureService.LocationApiVersion.LOCATIONMANAGER;
+        });
+
+        //start and bind the service
+        startAndBindGpsDataCaptureService();
+
+        //start capture data if the button state is START_CAPTURE, and stop if the state is
+        // STOP_CAPTURE
+        startAndStopButton.setOnClickListener((View v) -> {
+            if (startAndStopButtonState == Utils.ButtonState.START_CAPTURE) {
+                //hide radio group
+                apiRadioGroup.setVisibility(View.GONE);
+
+                showGpsDataAndStatusTextView();
+
+                startGpsCapture(locationApiVersion);
+
+                switchToStopButton();
+            } else {
+                stopGpsCapture(locationApiVersion);
+
+                hideGpsDataAndStatusTextView();
+
+                resetRadioGroup();
+
+                switchToStartButton();
+            }
+        });
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        startAndBindGpsDataCaptureService();
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        stopAndUnbindGpsDataCaptureService();
+        isBound = false;
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        startAndBindGpsDataCaptureService();
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        stopAndUnbindGpsDataCaptureService();
+        isBound = false;
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+    }
+
+    /**
+     * Set GPS if it's not enabled
+     */
+    private void setGpsEnabled() {
+        //Todo pop out a dialog to accept/deny enable setting??
+        Intent settingsIntent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+        startActivity(settingsIntent);
+    }
+
+    /**
+     * On click of Start button, start capturing gps data
+     */
+    public void startGpsCapture(GpsDataCaptureService.LocationApiVersion apiVersion) {
+        if (!isBound) {
+            Log.e(TAG, "GpsDataCaptureService is not bound, could not start capture.");
+            return;
+        }
+        Log.d(TAG, "Start capture data.");
+        gpsDataCaptureService.startCapture(apiVersion, gpsDataTextView, gpsStatusTextView);
```

Comment:
Can you please add check for  `Utils.isGpsEnabled(locationManager)` here  and show error on UI  if GPS is disabled?

Reply:
Done.